### PR TITLE
Tested Web Dashboard

### LINKED ISSUE #17: Test web dashboard
The web dashboard is lacking a large amount of tests. Add some abstractions to make testing easier and add tests for all the models. Attempt UI tests if possible

### PR DESCRIPTION
**Pull request by ghost**
closes #17

### CODE REVIEW

**Review by Camz-1601 — DISMISSED** [review]

**Review by OwenCross289 — DISMISSED** [review]
looks good

File: Dashboard.Tests/Dashboard.Tests.csproj [hunk 1]
```diff
@@ -19,6 +20,7 @@
         <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.4.0-2.final" />
         <PackageReference Include="Microsoft.CodeAnalysis.Razor" Version="6.0.10" />
         <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.1.0" />
+        <PackageReference Include="Moq" Version="4.18.2" />
```
Comment (OwenCross289) [inline Dashboard.Tests/Dashboard.Tests.csproj]: MOQ
MOQ

NSUB

**Review by laziestguy — DISMISSED** [review]
LGTM

**Review by Camz-1601 — DISMISSED** [review]

**Review by laziestguy — DISMISSED** [review]
Beautiful code here

**Review by OwenCross289 — APPROVED** [review]

**Review by laziestguy — APPROVED** [review]
Looking good!

**Pull request merged**